mesh_segmenter/src/mesh_segmenter.cpp

```diff
@@ -12,8 +12,25 @@
 
 #include <mesh_segmenter/mesh_segmenter.h>
 
+#include <log4cxx/basicconfigurator.h>
+#include <log4cxx/patternlayout.h>
+#include <log4cxx/consoleappender.h>
+
+log4cxx::LoggerPtr createConsoleLogger(const std::string& logger_name)
+{
+  using namespace log4cxx;
+  PatternLayoutPtr pattern_layout(new PatternLayout("[\%-5p] [\%c](L:\%L): \%m\%n"));
+  ConsoleAppenderPtr console_appender(new ConsoleAppender(pattern_layout));
+  log4cxx::LoggerPtr logger(Logger::getLogger(logger_name));
+  logger->addAppender(console_appender);
+  logger->setLevel(Level::getInfo());
+  return logger;
+}
+
 namespace mesh_segmenter
 {
+log4cxx::LoggerPtr MeshSegmenter::SEGMENTATION_LOGGER = createConsoleLogger("SegmentationLogger");
```

Comment:
I think this logger shouldn't be  exposed  to client code.  Could you move this to the source file instead?